encoder/vaapiencoder_h264.cpp

```diff
@@ -1142,10 +1326,17 @@ bool  VaapiEncoderH264::pictureReferenceListSet (
 
     for (i = 0; i < m_refList.size(); i++) {
         assert(picture->m_poc != m_refList[i]->m_poc);
-        if (picture->m_poc > m_refList[i]->m_poc)
+        if (picture->m_poc > m_refList[i]->m_poc
```

Comment:
Wired logical, could we put `if (picture->m_temporalId >= m_refList[i]->m_temporalId)` to a separate place?